### PR TITLE
Add regression tests for [TG-8299]

### PR DESCRIPTION
**Pull request by yumibagge**
- [x] Each commit message has a non-empty body, explaining why the change was made.
- n/a Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- n/a The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
- [x] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- n/a My commit message includes data points confirming performance improvements (if claimed).
- n/a My PR is restricted to a single feature or bugfix.
- n/a White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: 699c3f75e597da24144453097fccd387b5408589).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/117468740

File: jbmc/regression/jbmc/core-models-required/test-no-package.desc [hunk 1]
```diff
@@ -0,0 +1,9 @@
+KNOWNBUG
+HelloWorld.class
+--function "HelloWorld.helloWorld:()Ljava/lang/String;" --cp `../../../../scripts/format_classpath.sh . ../../../lib/java-models-library/target/core-models.jar` --context-include .
+^EXIT=0$
+^SIGNAL=0$
+--
+--
+`--context-include` with packageless terminate test-gen at `_Map_base::at`:
```
Comment (peterschrammel) [inline jbmc/regression/jbmc/core-models-required/test-no-package.desc]: Doesn't the same happen when running without `--cp ../../../../scripts/format_classpath.sh . ../../../lib/java-models-library/target/core-models.jar --context-include .` ? That would be a more general test case. Same for the other test case.

Comment (yumibagge) [reply jbmc/regression/jbmc/core-models-required/test-no-package.desc]: Unfortunately, the cbmc branch does not recognise `--context-include` at this moment, so I cannot guarantee, but when I run test-gen on your branch without models-library, it does not cause invariant violation nor this mysterious interruption at `_Map_base::at` so I assume models library is needed to reproduce these bugs.

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: 72aea6dd28c4c65d566193aa0c1cbd67cbaa518c).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/117492734

File: jbmc/regression/jbmc/core-models-required/test-no-package.desc [hunk 2]
```diff
@@ -0,0 +1,11 @@
+KNOWNBUG
+HelloWorld.class
+--function "HelloWorld.helloWorld:()Ljava/lang/String;" --context-include
+wrongpkg
+^EXIT=0$
+^SIGNAL=0$
+--
+--
+`--context-include` with non-existing package terminate test-gen at
```
Comment (peterschrammel) [inline jbmc/regression/jbmc/core-models-required/test-no-package.desc]:
```suggestion
`--context-include` with non-existing package terminate JBMC at
```

File: jbmc/regression/jbmc/core-models-required/test-no-package.desc [hunk 3]
```diff
@@ -0,0 +1,11 @@
+KNOWNBUG
+HelloWorld.class
+--function "HelloWorld.helloWorld:()Ljava/lang/String;" --context-include
+wrongpkg
```
Comment (peterschrammel) [inline jbmc/regression/jbmc/core-models-required/test-no-package.desc]: needs to be on the previous line

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: ba2e7083084fc060451a4b08ab6612f13ef6bd83).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/117569543

File: jbmc/regression/jbmc/core-models-required/test-no-package.desc (hunk 1 above)
Comment (yumibagge) [reply jbmc/regression/jbmc/core-models-required/test-no-package.desc]: @peterschrammel and I confirmed that `--context-include` contains wrong package, the issue is reproducible without the models path. However, `--context-include` points to a right package but not all necessary packages, test-gen (not JBMC) with models path causes invariant violation, which now is a separate PR https://github.com/diffblue/test-gen/pull/3616

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: 0dd25005aeaa5dbed737d1f4e9cf7ab56b5a702c).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/117607130

**Review by thk123 — APPROVED** [review]

**Review by peterschrammel — APPROVED** [review]

**Pull request merged**